syntax = "proto3";

service Booking {
  rpc GetListBookings(EmptyBooking) returns (stream BookingData) {}
  rpc GetBookingByUserId(UserId) returns (BookingFeedback) {}
  rpc AddBookingByUserId(OneBooking) returns (BookingFeedback) {}
}

message DateBooking {
  string date = 1;
  repeated string movies = 2;
}

message BookingData {
  string userid = 1;
  repeated DateBooking dates = 2;
}

message BookingFeedback {
  string message = 1;
  BookingData booking = 2;
}

message OneBooking {
  string userid = 1;
  string date = 2;
  string movieid = 3;
}

message UserId {
    string userid = 1;
}

message EmptyBooking {
}
